webrender/src/device/gfx/device.rs

```diff
@@ -2020,6 +2158,38 @@ impl<B: hal::Backend> Device<B> {
         }
     }
 
+    pub fn copy_cache_buffer(&mut self, dst: &Texture, src: &Texture) {
+        let ref src_buffer = self.gpu_cache_buffers[&src.id];
+        let ref dst_buffer = self.gpu_cache_buffers[&dst.id];
+        unsafe {
+            let cmd_buffer = self.command_pool[self.next_id].acquire_command_buffer();
+            cmd_buffer.begin();
+
+            let barriers = src_buffer
+                .transit(hal::buffer::Access::TRANSFER_READ)
```

Comment:
We are missing the `hal::buffer::Access::HOST_WRITE` bit to be transitioned to whenever we write into the buffer. That may explain the glitches you were seeing.

Reply:
Added it here: https://github.com/szeged/webrender/pull/297/commits/76b255b4b7ec31e659242e04dbd757177777abf4#diff-cdf03c96bd34cb98c2d09b0be7658232R2215